src/app/application/services/data-source/DataSourcePool.ts

```diff
@@ -6,7 +6,12 @@ import SharedPool from './SharedPool'
 export type Source = (params: Record<string, unknown>, source: {close: () => void}) => Promise<unknown>
 export type Sources = Record<string, Source>
 // reusable Type Utility for easy to use Types within Vue templates
-export type DataSourceResponse<T> = {data: T | undefined, error: Error | undefined, refresh: () => void}
+export type DataSourceResponse<T> = {
+  data: T | undefined
+  isLoading: boolean
```

Comment:
Did we never need this for table views?

Reply:
We did and we had this for all of them prior to migrating to DataSource/AppCollection. Instead, some of the views now use `data` being `undefined` as the signal for loading.

I think the loading state of data fetching should be indicated separately than via the variable that holds the data.

Reply:
> We did 

No we didn't.

We used the simple rules of:

- If we have no data then we are not 'connected' ...yet
- If we have no error then we are not 'erroneous' ...yet

Just to note, there will be a separate component that deals with data loading and showing visuals depending on the state depending on the above rules. DataSource is supposed to be very low level and with a very small API. For example here, with a polling or EventSource based DataSource `isLoading` is never reset to `false` meaning at the very least this state is confusingly named.

Another sidenote here is that as DataSources are theoretically never completely loaded they (or rather the component that will deal with those states) should have at least 4 states probably named something like `connecting`, `connected`, `error`, `disconnected`.